packages/chakra-components/src/MultiSelect/index.js

```diff
@@ -0,0 +1,743 @@
+  const placeholderColor = {
+    light: theme.colors.gray[400],
+    dark: theme.colors.whiteAlpha[400],
+  };
+
+  if (!values.length) {
+    return inputValue ? null : (
+      <PseudoBox
+        position="absolute"
+        top="50%"
+        transform="translateY(-50%)"
+        mx="2px"
+        color={placeholderColor[colorMode]}
+        {...props}
+      >
+        {children}
+      </PseudoBox>
+    );
+  }
+
+  return null;
+};
+
+MultiSelectPlaceholder.displayName = "MultiSelectPlaceholder";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const MultiSelectInput = forwardRef(
+  (
+    {
+      "aria-label": ariaLabel,
+      "aria-labelledby": ariaLabelledby,
+      isReadOnly,
+      isDisabled,
+      isInvalid,
+      isRequired,
+      ...props
+    },
+    ref,
+  ) => {
+    const { isInputHidden, getInputProps, isListBox } = useMultiSelectContext();
+
+    const ariaAttributes = {
+      "aria-autocomplete": "list",
+      "aria-label": ariaLabel,
+      "aria-labelledby": ariaLabelledby,
+    };
+
+    const { inputWrapperStyle, inputStyle } = useMultiSelectInputStyle({
+      isDisabled,
+      isInputHidden,
+      isListBox,
+    });
+
+    return (
+      <PseudoBox {...inputWrapperStyle} {...props}>
+        <AutosizeInput
+          type="text"
+          autoCapitalize="none"
+          autoComplete="off"
+          autoCorrect="off"
+          spellCheck="false"
+          inputStyle={inputStyle}
+          disabled={isDisabled}
+          {...ariaAttributes}
+          {...getInputProps()}
+        />
+      </PseudoBox>
+    );
+  },
+);
+
+MultiSelectInput.displayName = "MultiSelectInput";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const MultiSelectInputGroup = props => {
+  const {
+    isMulti,
+    placeholder,
+    renderCustomPlaceholder,
+    renderCustomSelectedOption,
+    renderCustomInput,
+  } = useMultiSelectContext();
+
+  const renderSelectedOption = () => {
+    if (isMulti) {
+      return (
+        <MultiSelectSelectedOption>
+          {({ selectedOption, handleOnClick }) => (
+            <Tag size="md" variant="solid" variantColor="blue">
+              <TagLabel>{selectedOption.label}</TagLabel>
+              <TagCloseButton tabIndex={-1} onClick={handleOnClick} />
+            </Tag>
+          )}
+        </MultiSelectSelectedOption>
+      );
+    }
+
+    return (
+      <MultiSelectSelectedOption>
+        {({ selectedOption }) => selectedOption.label}
+      </MultiSelectSelectedOption>
+    );
+  };
+
+  return (
+    <PseudoBox
+      position="relative"
+      display="flex"
+      alignItems="center"
+      flexWrap="wrap"
+      flex=" 1 1 0%"
+      px={2}
+      overflow="hidden"
+      minHeight="2.5rem"
+      {...props}
+    >
+      {renderCustomSelectedOption || renderSelectedOption()}
+      {renderCustomPlaceholder || (
```

Comment:
```suggestion
      {renderCustomPlaceholder(props) || (
```

I think this should have context of rendering exposed to parent. A case might be when you are showing placeholders like `Show 39 results`. Though this is not always used, this makes all renderCustom functions unified to be a `function` rather than just a JSX snippet.

Reply:
Not necessarily this should be a function, it can work even in the same way as a renderSelectedOption.

Reply:
The placeholder can now have the context. https://github.com/timelessco/chakra-components/pull/63/commits/899bebf988a919cf919be82fc52a86fd031eea6b
![image](https://user-images.githubusercontent.com/39694575/82317721-5ecc4b80-99ec-11ea-97cf-6f1ff3f6d1a7.png)